rust/src/core.rs

```diff
@@ -33,11 +34,39 @@ pub const APP_LAYER_EVENT_TYPE_PACKET      : i32 = 2;
 // From stream.h.
 pub const STREAM_START:    u8 = 0x01;
 pub const STREAM_EOF:      u8 = 0x02;
-pub const STREAM_TOSERVER: u8 = 0x04;
-pub const STREAM_TOCLIENT: u8 = 0x08;
 pub const STREAM_GAP:      u8 = 0x10;
 pub const STREAM_DEPTH:    u8 = 0x20;
 pub const STREAM_MIDSTREAM:u8 = 0x40;
+pub const DIR_TOSERVER:    u8 = 0b0000_0100;
+pub const DIR_TOCLIENT:    u8 = 0b0000_1000;
+
+#[repr(C)]
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Direction {
+    ToServer = 0x04,
+    ToClient = 0x08,
+}
+
+impl Default for Direction {
+    fn default() -> Self { Direction::ToServer }
+}
+
+impl From<u8> for Direction {
+    fn from(d: u8) -> Self {
+        if d == DIR_TOSERVER {
+            Direction::ToServer
+        } else if d == DIR_TOCLIENT {
+            Direction::ToClient
```

Comment:
I think we can drop this as its covered but the 2 following checks.

Reply:
Of course! :no_mouth: